api/v1beta1/openstackdataplaneservice_types.go

```diff
@@ -56,15 +56,37 @@ type OpenstackDataPlaneServiceCert struct {
 	EDPMRoleServiceName string `json:"edpmRoleServiceName,omitempty"`
 }
 
+// OpenStackDataPlaneConfigMap defines the properties of the ConfigMaps used
+// by the service.
+type OpenStackDataPlaneConfigMap struct {
+	// Name of the ConfigMap
+	Name string `json:"name"`
+
+	// Required is whether the ConfigMap is required to exist
+	// +kubebuilder:validation:Optional
+	Required bool `json:"required,omitempty"`
```

Comment:
We might want to make these references required by default as that is the more common usage of a reference.

Just be careful that adding a default true here will not work out of the box as omitempty will prevent the false value to be ever set.